### ISSUE TITLE
[Bug] Loss Weight Multiplier (w1 ≈ 200) Collides with WeightedRandomSampler to Starve Throttle and Brake

### ISSUE THREAD

**Issue by RustamSheoran** [Mohammedmt10/self-driving-model-in-simulation#7]
## Labels
`bug`, `critical`, `optimization`, `loss-function`

## Affected Files
- `model.py` (lines 239–268)
- `dataset.py` (lines 921–1016)

## Problem Summary
Task loss weights $w_1, w_2, w_3$ are calculated as the inverse variance/entropy of the **unweighted training set**. Because cars mostly drive straight, the unweighted steering variance is tiny ($\approx 0.005$), giving a huge multiplier $w_1 = 1 / 0.005 \approx 200$. However, the DataLoader samples batches using `WeightedRandomSampler`, which aggressively oversamples sharp turns.

## Root Cause Code
```python
# model.py lines 239-268
loss_weights = compute_loss_weights(train_list)
w1 = 1.0 / loss_weights["steer"]       # ~1 / 0.005 ≈ 200.0
w2 = 1.0 / loss_weights["throttle"]    # ~1 / 0.080 ≈ 12.5
w3 = 1.0 / loss_weights["brake"]       # ~1 / 0.200 ≈ 5.0

# But batches are sampled with WeightedRandomSampler!
sampler = WeightedRandomSampler(weights=sample_weights, num_samples=len(train_list), replacement=True)
train_loader = DataLoader(train_dataset, batch_size=BATCH_SIZE, sampler=sampler)
...
total = w1 * l_steer + w2 * l_throttle + w3 * l_brake
```

## Why It Is Completely Ruining Everything
In the actual resampled batches, turn oversampling inflates steering variance by 5x to 10x. Multiplying this already large steering loss by $w_1 \approx 200$ causes `w1 * l_steer` to contribute $6.0\text{--}10.0$ to the total loss, while throttle and brake contribute $\le 1.0$. Steering completely hijacks 90% of backpropagation gradients, starving the throttle and brake heads and causing `best_val_loss` checkpoint saving to track only steering performance while ignoring speed and stopping.

## Proposed Fix
Compute loss normalization baselines on the **resampled batch distribution**, or dynamically normalize task gradients using GradNorm or uncertainty weighting.
